### PR TITLE
Production Deployment (HTTP)

### LINKED ISSUE #144: production deployment (HTTP ONLY)
# Specs
* load balanced with 2 ec2 instances
* MongoDB atlas with replica set to ensure reliability + backups enabled
* AWS Hosting with HTTP only connections
* CI deployment triggered by any push to the main branch

# Associated Kanban Card
https://www.notion.so/uoftblueprint/Get-Repo-Deployed-for-staging-e1ae91193b19438e8f4607e183dbdfdb

### LINKED ISSUE #151: Frontend Speedup
Loading is really slow. To the degree that it's non-functioning.

https://tolgee.io/blog/2021/11/24/optimizing-react-to-the-bone
https://legacy.reactjs.org/docs/react-api.html

### PR DESCRIPTION
**Pull request by kenneth-miura**
# Description

* Add a button in frontend for testing, check database connectivity by creating a user and then displaying it
* set up Deployment via prod dockerfiles and prod docker composes
* See #144 associated kanban card for description of process
* Also fixed some linting errors that were causing issues in building the frontend.
Fixes #144 
Fixes #151 
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

File: .github/workflows/deploy_pipeline.yml
Comment (helen-li) [inline .github/workflows/deploy_pipeline.yml:1]: yom we love yaml files

**Review by helen-li — APPROVED** [review]
My one brain cell left understood maybe half of what's going on in the docker files but it builds and we trust

File: server/.dockerignore [hunk 1]
```diff
@@ -30,6 +30,6 @@ yarn-error.log
 .parallelperf.*
 .failed-tests
 TEST-results.xml
-tests
+test
```
[the remark this answers is not in the document]
Comment (kenneth-miura) [reply server/.dockerignore:33]: yeah, it's not needed for running the application. If we re setup automatic testing, it shouldn't need to use docker, so having this in .dockerignore is fine.

File: server/src/database/conn.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ interface collectionData {
 }
 
 const DB_CONN_STRING = process.env.DB_CONN_STRING || 'mongodb://mongo:27017'
-const DB_NAME = process.env.DB_NAME || 'default_db_name'
+const DB_NAME = "tces";
```
[the remark this answers is not in the document]
Comment (kenneth-miura) [reply server/src/database/conn.ts:36]: Ah yeah good catch, I did not realize I hardcoded this.

**Pull request merged**